ambersim/logger/logger.py

```diff
@@ -0,0 +1,146 @@
+import jax
+import wandb
+from torch.utils.tensorboard import SummaryWriter
+
+
+class BaseLogger:
+    """Base logger interface that defines common methods for logging metrics and parameters.
+
+    Attributes:
+        log_dir (str): Directory where logs are stored. If None, default locations are used.
+    """
+
+    def __init__(self, log_dir=None):
+        """Initializes the BaseLogger with a specified log directory.
+
+        Args:
+            log_dir (str): Directory to store the logs. If None, uses default log directory.
```

Comment:
If you add type hints in the function signatures, we can remove the redundant types in the docstring.